src/api/v4/client.js

```diff
@@ -163,7 +163,13 @@ Client.prototype.removeLayer = function (layer) {
  * @api
  */
 Client.prototype.removeLayers = function (layers) {
-  layers.forEach(this._removeLayer, this);
+  var clone = layers.slice(0);
```

Comment:
I have been thinking about what we did yesterday here and I think there are two slightly _easier_ solutions.

The one I think is better is to keep doing `.forEach` but using the cloned array to iterate, like:
```js
var layersToRemove = layers.slice(0);
layersToRemove.forEach(this._removeLayer, this);
```

And you will avoid the mutations on the original array passed on the parameters because of the cloned array. I think we tested this one but don't know why it didn't worked 🤔, I have just tested it on the browser and it works.

And the other one, avoids cloning the array, but it is riskier than the previous one:
```js
var length = layers.length;

for (var i = 0; i < length; i++) {
  this._removeLayer(layers[0]);
}
```

I prefer the first one because of obvious reasons. The thing that we did yesterday is not bad but I think it can be improved :)

Reply:
Changing the code to your first suggestion. :+1:

Reply:
💪